### PR TITLE
Feature/reordering dashboards

### PR DESCRIPTION
**Pull request by Cich0sza**

### CODE REVIEW

File: cogboard-webapp/src/components/BoardCard/index.js [hunk 1]
```diff
@@ -24,6 +28,51 @@ const BoardCard = ({ boardData, className }) => {
...
+      if (sourceIndex === targetIndex) {
+        return;
+      }
+
+      const { top, bottom } = ref.current.getBoundingClientRect();
+      const dropTargetMiddleY = bottom - (bottom - top) / 2;
+      const { y: dragSourceMouseY } = monitor.getClientOffset();
+
+      if (
+        (sourceIndex < targetIndex && dragSourceMouseY < dropTargetMiddleY) ||
```
Comment (szymon-owczarzak) [inline cogboard-webapp/src/components/BoardCard/index.js]: Can this complicated  `if` logic be named somehow and presented as a method ?

File: cogboard-webapp/src/components/BoardCard/index.js [hunk 2]
```diff
@@ -24,6 +28,51 @@ const BoardCard = ({ boardData, className }) => {
...
+  const ref = useRef(null);
+  const isLoggedIn = useSelector(({ app }) => !!app.jwToken);
+  const [{ isDragging }, drag] = useDrag({
+    item: { type: ItemTypes.BOARD, id, index },
+    canDrag: isLoggedIn,
+    collect: monitor => ({
+      isDragging: monitor.isDragging()
+    })
+  });
+  const [{ isOver }, drop] = useDrop({
```
Comment (szymon-owczarzak) [inline cogboard-webapp/src/components/BoardCard/index.js]: As a backend guy 4 return statements in one method are bit strange in my standards.

File: cogboard-webapp/src/components/BoardCard/index.js [hunk 3]
```diff
@@ -1,18 +1,22 @@
-import React from 'react';
+import React, { useRef } from 'react';
```
Comment (szymon-owczarzak) [inline cogboard-webapp/src/components/BoardCard/index.js]: nice I like that

File: cogboard-webapp/src/components/BoardCard/index.js [hunk 4]
```diff
@@ -59,13 +112,15 @@ const BoardCard = ({ boardData, className }) => {
               onClick={handleEditClick}
               aria-label="edit"
               size="small"
+              data-cy="board-card-edit-button"
```
Comment (szymon-owczarzak) [inline cogboard-webapp/src/components/BoardCard/index.js]: please be consistent with naming approach from your other PR:
`data-cy="edit-widget-cancel-button"`

File: cogboard-webapp/src/components/BoardCard/index.js (hunk 1 above)
Comment (Cich0sza) [reply cogboard-webapp/src/components/BoardCard/index.js]: done ;)

File: cogboard-webapp/src/components/BoardCard/index.js (hunk 2 above)
Comment (Cich0sza) [reply cogboard-webapp/src/components/BoardCard/index.js]: fixed

File: cogboard-webapp/src/components/DragCover/index.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import styled from '@emotion/styled/macro';
+
+const DragCard = ({ className }) => (
```
Comment (goeson00) [inline cogboard-webapp/src/components/DragCover/index.js]: Inconsistent name with the file name. Try to use already defined DStyledCard and modify properties there according to isDragging prop like in Widget component.

File: cogboard-webapp/src/components/BoardCard/index.js [hunk 6]
```diff
@@ -24,6 +28,49 @@ const BoardCard = ({ boardData, className }) => {
...
+
+  const checkIfPointerPassedMiddle = (sourceIndex, targetIndex, dragSourceMouseY, dropTargetMiddleY) => {
+    return (sourceIndex < targetIndex && dragSourceMouseY >= dropTargetMiddleY) ||
+          (sourceIndex > targetIndex && dragSourceMouseY <= dropTargetMiddleY)
+  }
+
+  const [{}, drop] = useDrop({
+    accept: ItemTypes.BOARD,
+    hover(item, monitor) {
+      if (ref.current) {
```
[the remark this answers is not in the document]
Comment (Cich0sza) [reply cogboard-webapp/src/components/BoardCard/index.js]: ok, I will use this approach.

File: cogboard-webapp/src/reducers/boards/allBoards.js [hunk 7]
```diff
@@ -15,6 +15,19 @@ const addBoard = (state, { payload }) => {
 const deleteBoard = (state, { payload: id }) =>
   state.filter(boardId => boardId !== id);
 
+const reorderBoards = (state, { payload }) => {
```
[the remark this answers is not in the document]
Comment (Cich0sza) [reply cogboard-webapp/src/reducers/boards/allBoards.js]: done :)

[the remark this answers is not in the document]
Comment (goeson00) [reply cogboard-webapp/src/reducers/boards/allBoards.js]: Closer but not exactly what I meant :) I meant replacing the whole reorderBoards and reorderWidgets functions with reorderItems function, cause they are exactly the same and operate on the array. I may be wrong, but is it possible?

File: cogboard-webapp/src/components/widgets/index.js [hunk 8]
```diff
@@ -55,7 +55,7 @@ const widgetTypes = {
     dialogFields: ['TimeZoneId', 'DateFormat', 'TimeFormat', 'DisplayDate', 'DisplayTime', 'DateTimeSize']
   },
   IFrameEmbedWidget: {
-    name: 'IFrame Embed widget',
+    name: 'Iframe Embed widget',
```
Comment (michalskrzypek) [inline cogboard-webapp/src/components/widgets/index.js]: Rename should be also applied to files names for better readability. I already done this on separate branch. You can safely delete the change from this branch 😀

Comment (Cich0sza) [reply cogboard-webapp/src/components/widgets/index.js]: done :)

File: cogboard-webapp/src/components/BoardCard/index.js [hunk 9]
```diff
@@ -40,7 +91,15 @@ const BoardCard = ({ boardData, className }) => {
...
-      <StyledCard onClick={handleBoardClick(id)}>
+      <StyledCard 
+        onClick={handleBoardClick(id)}
+        theme={theme}
+        isLoggedIn={isLoggedIn}
+        isDragging={isDragging}
+        isOver={isOver}
+        ref={ref}
+      >
+        {/* {isDragging ? <DragCover/>: null} */}
```
Comment (goeson00) [inline cogboard-webapp/src/components/BoardCard/index.js]: Remove this comment :)

**Pull request merged**